langchain4j-mcp/src/main/java/dev/langchain4j/mcp/McpToolProvider.java

```diff
@@ -0,0 +1,52 @@
+package dev.langchain4j.mcp;
+
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.mcp.client.McpClient;
+import dev.langchain4j.service.tool.ToolProvider;
+import dev.langchain4j.service.tool.ToolProviderRequest;
+import dev.langchain4j.service.tool.ToolProviderResult;
+import java.util.List;
+
+/**
+ * A tool provider backed by one or more MCP clients
+ */
+public class McpToolProvider implements ToolProvider {
+
+    private final List<McpClient> mcpClients;
+
+    private McpToolProvider(Builder builder) {
+        this.mcpClients = builder.mcpClients;
+    }
+
+    @Override
+    public ToolProviderResult provideTools(final ToolProviderRequest request) {
+        ToolProviderResult.Builder builder = ToolProviderResult.builder();
+        for (McpClient mcpClient : mcpClients) {
+            List<ToolSpecification> toolSpecifications = mcpClient.listTools();
```

Comment:
What if one of the calls fail? Should we fail completely or continue to call other servers?

Reply:
Maybe this could be configurable, WDYT?

Reply:
I made this configurable